### ISSUE TITLE
GRID_UNREGISTER_IF_STILL_DOWN_AFTER not propagating to nodes

### ISSUE THREAD

**Issue by jaymagpantay** [SeleniumHQ/docker-selenium#443]
## Meta -
Image(s):  
selenium/hub, node-chrome-debug

I'm running a selenium hub container with the following command:
docker run -d -e GRID_UNREGISTER_IF_STILL_DOWN_AFTER=300000 --name selenium-hub -p 4444:4444 selenium/hub

Afterwards, I launch a node-chrome-debug container. 

## Expected Behavior -
In the grid console, viewing the configuration machine should show unregisterIfStillDownAfter: 300000
## Actual Behavior -
It shows a value of 60000. I know the Dockerfile for the hub has a default value of 30000 for this, so I'm not sure where that 60000 is coming from.

**Comment by tparikka** [comment]
In looking at the [Dockerfile](https://github.com/tparikka/docker-selenium/blob/master/Hub/Dockerfile.txt) and the [generate_config](https://github.com/tparikka/docker-selenium/blob/master/Hub/generate_config) script for the hub, passing the env variable GRID_UNREGISTER_IF_STILL_DOWN_AFTER into the hub is correctly adding an attrib/val pair to the config file. However, looking at the sample JSON files for a [Node](https://github.com/SeleniumHQ/selenium/blob/master/java/server/src/org/openqa/grid/common/defaults/DefaultNodeWebDriver.json) and a [Hub](https://github.com/SeleniumHQ/selenium/blob/master/java/server/src/org/openqa/grid/common/defaults/DefaultHub.json) it seems as though unregisterIfStillDownAfter is a node-specific value that the hub would/should ignore and do nothing with. By comparison, the [Node generate_config](https://github.com/tparikka/docker-selenium/blob/master/NodeChrome/generate_config) doesn't declare a number of the Node env variables including unregisterIfStillDownAfter. @ddavison am I correct in thinking that we should take some of those Node-specific env variables out of the dockerfile and generate_config for Hub and expose them in the generate_config for the various nodes? If so I can write up a PR for it today, but let me know first in case there's something about Grid that I'm misunderstanding.

**Comment by ddavison** [comment]
you're absolutely right @tparikka , would definitely appreciate a PR!  thanks 👍

**Comment by tparikka** [comment]
@ddavison I have the PR started, and changes made to the Hub, NodeChrome, and NodeFirefox. While working on it I found that NodeChromeDebug, NodeFirefoxDebug, and NodePhantomJS lack a generate_config script but do have an entry_point.sh. In the case of the first two, are they inheriting the generate_config from their non-debug variants or would it be a separate feature request to add that support to the other docker images?

Also, is it reasonably possible to build via Cygwin or should I just suck it up and roll a Linux VM to build? I ran into an issue ( $'\r': command not found) while building in Cygwin and what I found online indicated it was an issue of the EOL encoding difference between *nix and Windows.
